src/seedu/addressbook/AddressBook.java

```diff
@@ -383,12 +418,175 @@ private static String executeCommand(String userInputString) {
+     *
+     * @param commandArgs  raw input from user
+     * @return message for successful editting of a person
+     */
+    private static String executeEditPerson(String commandArgs) {
+        Optional<String[]> person = checkValidPersonToEdit(commandArgs);
+        if (!person.isPresent()){
+            return "returning to edit screen";
+        }
+        final String[] personToAdd = rebuildPerson(person.get());
+        return getMessageForSuccessfulEditPerson(personToAdd);
+        
+	}
+    
+    /**
+     * Shows details about the person successfully editted 
+     *
+     * @param addedPerson - String[] representation of a person
+     * @return formatted string to declare a person has been successfully added
+     */
+	private static String getMessageForSuccessfulEditPerson(String[] addedPerson) {
+        return String.format(MESSAGE_EDITED,
+                getNameFromPerson(addedPerson), getPhoneFromPerson(addedPerson), getEmailFromPerson(addedPerson), getSchoolFromPerson(addedPerson));
+	}
+
+    /**
+     * edits a persons 4 values, name, phone, email and school individually by taking input from a person 
+     *
+     * @param person - String[] representation of a person
+     * @return person - String[] representation of the new details of the person
+     */
+	private static String[] rebuildPerson(String[] person) {
+        showToUser(person);
+        final String[] personToAdd = new String[4];
+        personToAdd[0] = replaceUserDetails(person,"name",MESSAGE_TO_EDIT_PERSON_NAME,0);
+        personToAdd[1] = replaceUserDetails(person,"phone",MESSAGE_TO_EDIT_PERSON_PHONE,1);
+        personToAdd[2] = replaceUserDetails(person,"email",MESSAGE_TO_EDIT_PERSON_EMAIL,2);
+        personToAdd[3] = replaceUserDetails(person,"school",MESSAGE_TO_EDIT_PERSON_SCHOOL,3);
+        ALL_PERSONS.remove(person);     
+        addPersonToAddressBook(personToAdd);
+		return personToAdd;
+	}
+	
+    /**
+     * replaces a specific variable of a person, taking in user input
+     *
+     * @param person - String[] representation of a person, property - the specific name of the property to be changed
+     *                 help_message - the message to show the user when adding this variable, and finally an int type
+     *                 which is the type of variable it is - the type allows an automatic checking of the validity
+     *                 of the user input using the function  checkValidProperty(userInput,type)
+     *                 
+     * @return person - String[] representation of the new details of the person
+     */
+	private static String replaceUserDetails(String[] person, String property, String help_message,int type) {
+		final String tempProperty;
+		showToUser(help_message);
+        String userInput = getEditUserInput();
+        while (!checkValidProperty(userInput,type)&&!userInput.equals("")){
+        	showToUser("Please Enter Valid " + property);
```

Comment:
please use 4 spaces for indentation, as required by the coding standards.
Tip: you can get eclipse to do that for you, check out how!